### DISCUSSION TITLE
Craft [4.x] : How to register the user if the email address already used in guest cart.

### DISCUSSION THREAD

**Discussion by aodihis** [craftcms/cms#11028] [Q&A] [answered]
I am not sure, where to ask for this one, in Commerce or in Craft cms it self, but here is the scenario.

1. Make a purchase in craft commerce as quest using email test@example.com, so the craft will created this email as a new user element.
2. After that the customer want to register that email, we do it using craft endpoint `users/save-users`

In step 2 the registration user will fail because the email already used before.
So in that case, is there are way to register the user with the email address that have been used in cart as quest ?

**Comment by brandonkelly** [comment, marked as answer]
We overlooked the need to publicly-register an account that was already created as an inactive account. That’s been fixed for the next Craft 4 release (d3f7bf4ff400c8386de2595ce23663119e0400a6).

**Reply by aodihis** [reply]
@brandonkelly, Thank For the fix, tested on RC2 and it works like a charm. Sorry for my little confusion before.

**Reply by brandonkelly** [reply]
@aodihis Was going to reply once we got RC2 + Commerce RC1 out (any minute now…). Your originally comment was correct, but there was actually a separate bug, which was also inadvertently fixed by my commit. And one more related one that’s about to be fixed by Commerce RC1 (users registered on order complete aren’t getting usernames set).

Glad the fix worked for you though!
